test/hotspot/jtreg/compiler/arraycopy/TestSunkLoadAntiDependency.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2025, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8341976
+ * @summary C2: use_mem_state != load->find_exact_control(load->in(0)) assert failure
+ * @run main/othervm -XX:-TieredCompilation -XX:-BackgroundCompilation -XX:-UseOnStackReplacement
+ *                    -XX:CompileOnly=TestSunkLoadAntiDependency::test1 TestSunkLoadAntiDependency
```

Comment:
Drive by comments: 
Is `-XX:-UseOnStackReplacement` required to reproduce the issue?

There was also a crash when running with `-XX:+TraceLoopOpts`. Can you also add a run with that flag to verify that this patch also fixes that?

Reply:
> Drive by comments: Is `-XX:-UseOnStackReplacement` required to reproduce the issue?

No, it's not. I trimmed the list of options for the tests.

> There was also a crash when running with `-XX:+TraceLoopOpts`. Can you also add a run with that flag to verify that this patch also fixes that?

Added. The `TraceLoopOpts` crash reproduces: the code hits a malformed counted loop. I tweaked the printing code.

Reply:
It doesn't look an actual issue to me. `PhiNode::Value` manages to narrow the trip `phi`'s type of the pre loop enough that it's a constant. So the loop no longer has the expected counted loop shape but the loop exit condition that should constant fold doesn't because it's guarded by an `Opaque1` node.

Reply:
This sounds very similar to [JDK-8297752](https://bugs.openjdk.org/browse/JDK-8297752) which also caused problems outside of `TraceLoopOpts`. We should probably handle this separately from this PR.

Reply:
Done.

Reply:
Thanks!